// react-auth-context/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _hoistNonReactStatics = require('hoist-non-react-statics');

var _hoistNonReactStatics2 = _interopRequireDefault(_hoistNonReactStatics);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getDisplayName = function getDisplayName(Component) {
  return Component.displayName || Component.name || 'Component';
};

function injectContext(AuthComponent) {
  /* eslint-disable react/prefer-es6-class */
  var ComponentWithAuthContext = _react2.default.createClass({
    displayName: 'ComponentWithAuthContext',

    contextTypes: {
      isLoggedIn: _react.PropTypes.bool,
      isLoggingIn: _react.PropTypes.bool,
      currentUser: _react.PropTypes.any
    },

    render: function render() {
      var context = {
        isLoggedIn: this.context.isLoggedIn,
        isLoggingIn: this.context.isLoggingIn,
        currentUser: this.context.currentUser
      };

      return _react2.default.createElement(AuthComponent, (0, _extends3.default)({}, context, this.props));
    }
  });

  ComponentWithAuthContext.displayName = 'WithAuthContext(' + getDisplayName(AuthComponent) + ')';
  return (0, _hoistNonReactStatics2.default)(ComponentWithAuthContext, AuthComponent);
}

exports.default = injectContext;
